### PR TITLE
[RFR] Replace component injection with element injection

### PR DESCRIPTION
**Pull request by djhi**
Revert #3262

### CODE REVIEW

**Comment by djhi** [comment]
It's green but I still have a strange warning left to fix

File: packages/ra-ui-materialui/src/list/List.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
-/* eslint no-console: ["error", { allow: ["warn", "error"] }] */
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/List.js]: can this be the cause of your warning?

Comment (djhi) [reply packages/ra-ui-materialui/src/list/List.js]: Nope, it's a warning about react-router

**Comment by djhi** [comment]
> There were 100 files changed in the previous PR, only 91 in this PR. Some changes may be missing.

Yes some changes are still relevant (typos, etc.)

**Comment by fzaninotto** [comment]
Thanks a lot!

**Pull request merged**